src/main/resources/public/jsx/profile/profile.jsx.js

```diff
@@ -1,67 +1,121 @@
 'use strict';
 
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React from "react";
+import ReactDOM from "react-dom";
 import renderIf from "render-if";
+import "../csrf";
+import "../my";
+import t from "../util/message";
 
-const moment = require('moment');
-
-import '../csrf';
-import '../my';
-import t from '../util/message';
+import {Form, InputText, Select, SubmitButton, InputDatePicker} from "../util/form";
+import {GeoAreaAutosuggest} from "../util/geoarea-autosuggest.jsx";
 
-import { Form, InputText, Select, SubmitButton } from '../util/form'
+const moment = require('moment');
 
 class Profile extends React.Component {
     state = {
         userProfile: {
             nickname: '',
             email_address: '',
-            locale: ''
+            locale: '',
+            given_name: '',
+            middle_name: '',
+            family_name: '',
+            phone_number: '',
+            gender: '',
         },
+        genders: ['male','female'],
         languages: [],
-        passwordChangeEndpoint: ''
+        passwordChangeEndpoint: '',
     }
     componentDidMount() {
         $.ajax({
             url: profile_service
         })
-        .done(data => this.setState({ userProfile: data.userProfile, languages: data.languages, passwordChangeEndpoint: data.passwordChangeEndpoint }))
+        .done(
+            function (data) {
+                if (data.userProfile.birthdate) {
+                    // console.log("brith - 1")
+                    let dayofbirth = data.userProfile.birthdate
+                    dayofbirth[1] = dayofbirth[1] - 1
+                    data.userProfile.birthdate = dayofbirth
+                } else {
+                    data.userProfile.birthdate = undefined
+                }
+                this.setState({ userProfile: data.userProfile, languages: data.languages, passwordChangeEndpoint: data.passwordChangeEndpoint })
+            }.bind(this)
+            //data => this.setState({ userProfile: data.userProfile, languages: data.languages, passwordChangeEndpoint: data.passwordChangeEndpoint })
```

Comment:
Don't leave commented out code, I let you check other places where there are some.